substrate-archive-common/src/error.rs

```diff
@@ -16,7 +16,7 @@
 use std::{env, io};
 use thiserror::Error;
 
-pub type Result<T> = std::result::Result<T, Error>;
+pub type Result<T, E = Error> = std::result::Result<T, E>;
```

Comment:
This change doesn't make sense to me, either make it specific to this module or use the generic Result from the standard library.

//cc @insipx

Reply:
I don't believe it to be strictly necessary either, I just see it as an alternative way to represent the `Result` that was already being used.

That being said, I don't see anything wrong with it perse, and don't have strong opinions either way